safe-global/web-core
# fix: pressing enter doesn't set max

## What it solves

Resolves #1164

## How this PR fixes it

The `InputValueHelper` required the `type="button"` attribute to prevent it from submitting the parent form and `shouldValidate` to be `true` in `onMaxAmountClick`. I also wrapped `InputValueHelper` in the correct `InputAdornment` component.

## How to test it

Attempt to send tokens and, after selecting one and entering an amount, press the enter key whilst the field is in focus. Observe no change in value or form submission. Doing so in an empty field should also result in the same.

Clicking "Max" should set the max value, as well as validate the amount, e.g. when no token amount exists, it should show an error regarding a value of `0`.
